mlflow/entities/span.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+import json
+from dataclasses import asdict, dataclass, field
+from typing import Any, Dict, List, Optional, Union
+
+from mlflow.entities._mlflow_object import _MLflowObject
+from mlflow.entities.span_context import SpanContext
+from mlflow.entities.span_event import SpanEvent
+from mlflow.entities.span_status import SpanStatus
+
+
+@dataclass
+class Span(_MLflowObject):
+    """A span object. (TODO: Add conceptual guide for span vs trace.)
+
+    Args:
+        name: Name of the span.
+        context: SpanContext object that contains the request_id (= trace_id) and span_id.
+        parent_span_id: Id of the parent span. If None, the span is the root span.
+        span_type: Type of the span. Can be a pre-defined enum or a custom string.
+        status: Status of the span.
+        start_time: Start time of the span in microseconds.
+        end_time: End time of the span in microseconds.
+        inputs: Inputs data of the span. Optional.
+        outputs: Outputs data of the span. Optional.
+        attributes: Arbitrary key-value pairs of the span attributes. Optional.
+        events: List of events that happened during the span. Optional.
+    """
+
+    name: str
+    context: SpanContext
+    parent_span_id: Optional[str]
+    span_type: Union[SpanType, str]
```

Comment:
```suggestion
    span_type: str
```

Can this be `str`?

Reply:
the motivation was to explicitly put `SpanType` enum to docstring so users can realize there are a set of pre-defined types.. do you think it's a bit confusing?

Reply:
I think having both is valuable.

Reply:
It's a bit confusing to me. Another concern is `SpanType` looks like it takes an instance of `SpanType`.

Reply:
What about making type as `str`, and add default value like `= SpanType.UNKNOWN`? This way users can tell there is some predefined set.

Reply:
that sounds good!